exercises/list-ops/canonical-data.json

```diff
@@ -0,0 +1,147 @@
+{
+  "#": [
+    "Though there are no specifications here for dealing with large lists,",
+    "implementers may add tests for handling large lists to ensure that the",
+    "solutions have thought about performance concerns."
+  ],
+  "append": {
+    "description": "append entries to a list and return the new list",
+    "cases": [
+      {
+        "description": "empty lists",
+        "list1": [],
+        "list2": [],
+        "expected": []
+      },
+      {
+        "description": "empty list to list",
+        "list1": [],
+        "list2": [1,2,3,4],
+        "expected": [1,2,3,4]
+      },
+      {
+        "description": "non-empty lists",
+        "list1": [1,2],
+        "list2": [2,3,4,5],
+        "expected": [1,2,2,3,4,5]
+      }
+    ]
+  },
+  "concat": {
+    "description": "concat lists and lists of list into a new list",
+    "cases": [
+      {
+        "description": "empty list",
+        "lists": [],
+        "expected": []
+      },
+      {
+        "description": "list of lists",
+        "lists": [[1,2],[3],[],[4,[5,6]]],
```

Comment:
notice that the `[4, [5, 6]]` is a array of either ints or array of ints. This will require some unfortunate contortions in languages where arrays (or the analogous type) may only contain values of only one type. In general, one might think concatenate would take an `[[a]]` and return an `[a]`. If the inner array may be arbitrarily nested, given the expected result, this operation is `concat + flatten`, not just concat - is this desirable? Is there, indeed, a language where the operation named `concat` does also flatten (and can you tell me which)?

I would recommend not expecting this function to flatten.